src/dataset-builder/DatasetBuilderUtils.ts

```diff
@@ -21,72 +27,17 @@ export interface Criteria {
   option: SnapshotBuilderOption;
 }
 
-/** API types represent the data of UI types in the format expected by the backend.
- * They are generally subsets or mappings of the UI types. */
-
-export interface CriteriaApi {
-  // This is the ID for either the domain or the program data option
-  id: number;
-  kind: SnapshotBuilderOptionTypeNames;
-  name: string;
-  count?: number;
-}
-
-export interface DomainCriteriaApi extends CriteriaApi {
-  kind: 'domain';
-  // This is the id for the selected concept
-  conceptId: number;
-}
-
-export interface ProgramDataRangeCriteriaApi extends CriteriaApi {
-  kind: 'range';
-  low: number;
-  high: number;
-}
-
-export interface ProgramDataListCriteriaApi extends CriteriaApi {
-  kind: 'list';
-  values: number[];
-}
-
-export type AnyCriteriaApi = DomainCriteriaApi | ProgramDataRangeCriteriaApi | ProgramDataListCriteriaApi;
-
-export interface CriteriaGroupApi {
-  name: string;
-  criteria: AnyCriteriaApi[];
-  mustMeet: boolean;
-  meetAll: boolean;
-}
-
-export interface CohortApi extends DatasetBuilderType {
-  criteriaGroups: CriteriaGroupApi[];
-}
-
-export type ValueSetApi = {
-  name: string;
-  values: string[];
-};
-
-export type DatasetRequestApi = {
-  cohorts: CohortApi[];
-  conceptSets: ConceptSet[];
-  valueSets: ValueSetApi[];
-};
-
-export type DatasetAccessRequestApi = {
-  name: string;
-  researchPurposeStatement: string;
-  datasetRequest: DatasetRequestApi;
-};
-
 /** Below are the UI types */
 export interface DomainCriteria extends Criteria {
   kind: 'domain';
   conceptId: number;
   conceptName: string;
-  option: SnapshotBuilderDomainOption;
+  option: DomainOption;
```

Comment:
Why? we need to include `conceptCount` and `participantCount`

Reply:
I think this was my suggestion, since these are used on the details page I think they should be required in the API types and we don't need another type here.

Also, looking at the code that uses these fields (`TileDisplay` in `DatasetBuilderDetails.ts`) it is still referring to the API types, not these new types. I think the reason why this isn't flagged as an error is because these fields are optional in the downstream type (`DomainDisplayProps.displayInformation`). Making these types non-optional should expose the issue.